compiler/backend/src/org/jetbrains/kotlin/codegen/inline/ParametersBuilder.kt

```diff
@@ -92,10 +89,10 @@ class ParametersBuilder private constructor() {
 
     fun buildParameters(): Parameters {
         var nextDeclarationIndex = (params.maxOfOrNull { it.declarationIndex } ?: -1) + 1
-
+        var nextOffset = nextParameterOffset
         return Parameters(params.map { param ->
             if (param is CapturedParamInfo) {
-                param.cloneWithNewDeclarationIndex(nextDeclarationIndex++)
+                param.cloneWithIndices(nextOffset.also { nextOffset += param.type.size }, nextDeclarationIndex++)
```

Comment:
Looks like it's also worth to have tests with Long/Double vals

Reply:
Added an extra long to defaultAfterCapturing.kt; the expected offset for the capture of O is now 2+1+1=4.